### DISCUSSION TITLE
something erro with submission about pbs

### DISCUSSION THREAD

**Discussion by LZH-1996** [deepmodeling/dpgen#914] [Q&A] [answered]
What is the reason for this error when I use the "dpgen run param.json machine.json " to submit the task to the cluster's **pbs** queue (i confirm the dp-kit can run and i think there is a problem with the generated script, but not know why)
<img width="885" alt="image" src="https://user-images.githubusercontent.com/46983771/186349210-865f7a99-2c9e-4f37-b208-dda47eaae244.png">
<img width="696" alt="image" src="https://user-images.githubusercontent.com/46983771/186349432-5978515d-08c5-46fe-9457-fdb8b4f1b113.png">
<img width="539" alt="image" src="https://user-images.githubusercontent.com/46983771/186349582-ce3f42ed-b399-4b95-b109-688f010c6e1f.png">

**Comment by HuangJiameng** [comment]
<img width="754" alt="image" src="https://user-images.githubusercontent.com/105633685/186394076-d11f8715-5501-4091-8c6c-b594a2f2907f.png">
It is about [dpdispatcher](github.com/deepmodeling/dpdispatcher)  
@felix5572 Could you please take a look? Thx

**Reply by hahha2022** [reply]
你的PBS 版本太低了，导致执行pbs.py这个模块时，status_word这个变量得不到正确的值（R,Q,C,E）
解决办法 1. 换个队列系统
2. 把status_word 这个变量打印出来，然后用split方法切割一下，使其得到正确的值。

**Comment by hahha2022** [comment, marked as answer]
你的PBS 版本太低了，导致执行pbs.py这个模块时，status_word这个变量得不到正确的值（R,Q,C,E）
解决办法 1. 换个队列系统
2. 把status_word 这个变量打印出来，然后用split方法切割一下，使其得到正确的值。

**Reply by HuangJiameng** [reply]
> 你的PBS 版本太低了，导致执行pbs.py这个模块时，status_word这个变量得不到正确的值（R,Q,C,E） 解决办法 1. 换个队列系统 2. 把status_word 这个变量打印出来，然后用split方法切割一下，使其得到正确的值。

Thanks for your reply~ Kindly reminder: it is recommended to use English for everyone to understand. 
Here are the same contents as above: 
Your PBS version is too low, so when the pbs.py module is executed, the variable `status_word` cannot get the correct value (R, Q, C, E)
Possible solutions:
1. Change the queue system
2. Print the variable status_word, and then use `split` to cut it to get the correct value.

**Reply by LZH-1996** [reply]
Thanks for yours repl , I have been resolved (change the ”PBS“ to  “Torque” in the dpdispatcher.pbs module)

**Comment by Project-M-ZQ** [comment]
May I ask how the author solved this problem specifically, I also encountered the same situation

**Reply by LZH-1996** [reply]
You can read the previous reply, you  can try to change the batch_type from “PBS” to “Torque”, if you are the pbs task system

**Comment by njzjz** [comment]
What is your PBS version?

**Reply by LZH-1996** [reply]
I don't know how to query the version, after read the dpdispatcher manual, I changed the batch type to "Torque" and it worked.
![121](https://user-images.githubusercontent.com/46983771/191260448-d93eaf8c-60e3-43a6-9b1b-960951565142.PNG)
![pbs](https://user-images.githubusercontent.com/46983771/191260459-5ff67990-c0ab-4e55-9507-825c39273a4a.PNG)

**Comment by Project-M-ZQ** [comment]
Sorry, I was using a public machine, I don't know how to check the PBS version




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/dpgen"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月20日(星期二) 下午2:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/dpgen] something erro with submission about pbs (Discussion #914)





 
What is your PBS version?

**Reply by hahha2022** [reply]
Please see my answer below

**Comment by hahha2022** [comment]
I am sorry , i am poor english. But one may solve this problem if follow the steps below:



First, open the file that named "pbs.py", go to the line 67 ，that is, the number of lines reported by the system

you will find the following codes:      
```
        ......
        status_line = stdout.read().decode('utf-8').split ('\n')[-2]
        status_word = status_line.split ()[-2]        
        ......
```
The error happend when the variable "status_word" cannot get the correct value (R, Q, C, E...) , due to the old PBS version. Therefore, we could changed the code like this:
```
       ……
        status_line = stdout.read().decode('utf-8').split ('\n')[-2]
        print("\n\n\n the value of status_line is \n %s \n\n\n" %status_line)
        #status_word = status_line.split ()[-2]        
        ……
```
After changing,  run the dpgen again. you will get a message like this:
```
        ……
        the value of status_line is
        **********balabalabalabalabala<R>balabalabalabalabala
         ……
```

Note: "************************* balababalabala<R>balababalabala" in here is a general information. You may get the different printed message because the different PBS version.  What we need to do is to use the basic Python syntax to cut out the "R" in the above message and pass it to the "status_ word" variable.


For example, the version of my PBS is 2.6, my information printed is:

_status_line is 
.... </Job_Id<Job_Name>da102a4f896f4d9eb287d847f331755811268906.sub</Job_Name<Job_Owner>liu103@host</Job_Owner><job_state>R</job_state><queue>sma</queue><server>host</server><Checkpoint>u</Checkpoint><ctime>1664182223</ctime><Error_Path> ...._



So on line 67, my code was changed to  :
```
        ....
        status_line = stdout.read().decode('utf-8').split ('\n')[-2]
        print("\n\n\nstatus_line is %s \n\n\n" %status_line)      
        status_word = status_line.split('<job_state>')[1].split('</job_state>')[0]
        ....
```
In the above modification, I used the "split" method in Python to cut the above string "status_line ", so as to make the "status_ word" variable gets correct value


It is work in my computer, I hope it can help you!!
